nagios/bin/pmp-check-mysql-deleted-files

```diff
@@ -122,22 +124,23 @@ mysql_exec() {
 # ########################################################################
 _lsof() {
    PATH="$PATH:/usr/sbin:/sbin"
-   if ! lsof -p $1 2>/dev/null; then
-      if ! /bin/ls -l /proc/$1/fd 2>/dev/null; then
-         fstat -p $1 2>/dev/null
+   if ! ${OPT_SUDO:+sudo }lsof -p $1 2>/dev/null; then
+      if ! ${OPT_SUDO:+sudo }/bin/ls -l /proc/$1/fd 2>/dev/null; then
+         ${OPT_SUDO:+sudo }fstat -p $1 2>/dev/null
       fi
    fi
 }
 
 # ########################################################################
 # Generate a list of file handles that MySQL has open, but which are deleted,
-# and are not temp files such as /tmp/ib* files (InnoDB) or /tmp/ML* files
-# (binary logging).  The first argument is a file containing the output of lsof
-# or ls -l for the open files.  The second argument is the server's tmpdir.
+# and are not in the exclude list, such as /tmp/ib* files (InnoDB) or /tmp/ML*
+# files (binary logging).  The first argument is a file containing the output
+# of lsof or ls -l for the open files.  The second argument is an exclude list,
+# which always includes the server's tmpdir.
 # ########################################################################
 check_deleted_files() {
-   awk -v tmpdir="${2}" '
-   /\(deleted\)/ { if ( index($0, tmpdir) == 0 ) {
+   awk -v exclude=${2} '
```

Comment:
You have removed double-quotes, please add them back

Reply:
Don't remember why I removed them, but quotes are better than injection errors.  I'll add those back.